apps/server/src/modules/board/service/submission-item.service.spec.ts

```diff
@@ -92,4 +97,49 @@ describe(SubmissionItemService.name, () => {
 			});
 		});
 	});
+
+	describe('update', () => {
+		const setup = () => {
+			const submissionContainer = submissionContainerElementFactory.build();
+			const submissionItem = submissionItemFactory.build();
+
+			boardDoRepo.findParentOfId.mockResolvedValueOnce(submissionContainer);
+
+			return { submissionContainer, submissionItem };
+		};
+
+		it('should fetch the SubmissionContainer parent', async () => {
+			const { submissionItem } = setup();
+
+			await service.update(submissionItem, true);
+
+			expect(boardDoRepo.findParentOfId).toHaveBeenCalledWith(submissionItem.id);
+		});
+
+		it('should call bord repo to save submission item', async () => {
+			const { submissionItem, submissionContainer } = setup();
+
+			await service.update(submissionItem, true);
+
+			expect(boardDoRepo.save).toHaveBeenCalledWith(submissionItem, submissionContainer);
+		});
+
+		it('should save completion', async () => {
+			const { submissionItem, submissionContainer } = setup();
+			submissionItem.completed = false;
+			await service.update(submissionItem, false);
+
+			expect(boardDoRepo.save).toHaveBeenCalledWith(submissionItem, submissionContainer);
```

Comment:
this test cannot be falsified for some reason
expect(boardDoRepo.save).toHaveBeenCalledWith(expect partial submissionItem { completed: false }, submissionContainer);

Reply:
done